pinot-core/src/main/java/org/apache/pinot/core/realtime/converter/stats/RealtimeNoDictionaryColStatistics.java

```diff
@@ -63,12 +68,52 @@ public int getCardinality() {
 
   @Override
   public int getLengthOfShortestElement() {
-    return lengthOfDataType(); // Only fixed length data types supported.
+    FieldSpec.DataType dataType = _blockValSet.getValueType();
+    if (dataType == FieldSpec.DataType.STRING || dataType == FieldSpec.DataType.BYTES) {
+      // variable width no dictionary columns
+      int minLength = Integer.MAX_VALUE;
+      BaseSingleColumnSingleValueReaderWriter readerWriter = (BaseSingleColumnSingleValueReaderWriter)_forwardIndex;
```

Comment:
Can we keep track of shortest and longest element in the fwd index and just read it here? Will save time as well as garbage generation during segment build.

We can definitely compute min and max in one go, and not need to walk over the fwd index for each of them separately.

Reply:
Done